src/lock/mod.rs

```diff
@@ -77,10 +77,20 @@ pub fn config(ctx: &Context, config: Config) -> Result<LockedConfig> {
 
     // Create a map of unique `Source` to `Vec<Plugin>`
     let mut map = IndexMap::new();
+    let profile = std::env::var("SHELDON_PROFILE");
     for (index, plugin) in externals {
```

Comment:
I think we should allow inline plugins to have profiles too. So you will need to filter `inlines` as well